forge-game/src/main/java/forge/game/zone/Zone.java

```diff
@@ -214,7 +214,26 @@ public final boolean isCardAddedThisTurn(final Card card, final ZoneType origin)
         if (!cardsAddedThisTurn.containsKey(origin)) {
             return false;
         }
-        return cardsAddedThisTurn.get(origin).contains(card);
+        if (cardsAddedThisTurn.get(origin).contains(card)) {
+            long ts = getCardsAddedThisTurn(origin).get(getCardsAddedThisTurn(origin).lastIndexOf(card)).getTimestamp();
```

Comment:
First i don't like that the get Function is called 4+ times

Then i need to check where this function is called from, so the code below can be optimized

Aren't we just looking for if card has the same time stamp as in the list?

Reply:
Only called once from CardProperty

`equalsWithTimestamp` doesn't work because it stores the LKI of the card before the zone change
has to be that way though so cards like _Ashen-Skin Zubera_ always count correctly